microsoft/testsuites/network/networksettings.py

```diff
@@ -42,9 +42,31 @@ def validate_ringbuffer_settings_change(self, node: Node) -> None:
             original_rx = int(interface_settings.current_ring_buffer_settings["RX"])
             original_tx = int(interface_settings.current_ring_buffer_settings["TX"])
 
-            # selecting lesser values to avoid crossing max values of Rx and Tx.
-            expected_rx = original_rx - 2
-            expected_tx = original_tx - 2
+            # In the netvsc driver code, default sizes are defined like below -
+            # Recieve Buffer, RX
+            # NETVSC_RECEIVE_BUFFER_DEFAULT = (1024 * 1024 * 16)
+            # NETVSC_RECV_SECTION_SIZE = 1728
+            #
+            # Send Buffer, TX
+            # NETVSC_SEND_BUFFER_DEFAULT =  (1024 * 1024 * 1)
+            # NETVSC_SEND_SECTION_SIZE = 6144
+            original_rxbuffer = round((original_rx * 1728) / (1024 * 1024))
+            original_txbuffer = round((original_tx * 6144) / (1024 * 1024))
+
+            rxbuffer = (
+                (original_rxbuffer - 2)
+                if original_rxbuffer - 2 > 0
+                else (original_rxbuffer + 2)
+            )
+
+            txbuffer = (
+                (original_txbuffer - 2)
+                if original_txbuffer - 2 > 0
+                else (original_txbuffer + 2)
+            )
+
+            expected_rx = int((rxbuffer * 1024 * 1024) / 1728)
+            expected_tx = int((txbuffer * 1024 * 1024) / 1728)
```

Comment:
The original number is 6144 above, is it correct here?

Reply:
Thanks. Good catch